### DISCUSSION TITLE
projects-droidtale

### DISCUSSION THREAD

**Discussion by giscus** [MrPowerGamerBR/PersonalWebsiteComments#9] [Fan Comments]
# projects-droidtale

Droidtale (Android + Undertale, get it?) is a very "hacky" Undertale port for Android devices.

https://mrpowergamerbr.com/us/projects/droidtale/

**Comment by asda488** [comment]
I know it's been ages since this tutorial, but again, thank you very much for this!

**Reply by MrPowerGamerBR** [reply]
:3

**Comment by pokeyjojo** [comment]
the game works perfectly! however even after setting a macro (using game keyboard plus) none of the buttons open the menu! what should i do?

**Reply by pokeyjojo** [reply]
fixed it! seems i had a comma before the macro started. thank you!

**Comment by LordFonDragon** [comment]
Can I use this method to create a port of a mod (just changing data.win file, or that wouldn't work at all?

**Reply by MrPowerGamerBR** [reply]
Yes, it would work :3

**Comment by TrexCM241** [comment]
Hi!
I tried to download undertale version 1.08c, but the soundtrack doesn't sound, just the sound of talking, please help me to fix this.

**Reply by TrexCM241** [reply]
Andddd,is this only work 1.05a down?

**Reply by MrPowerGamerBR** [reply]
If the soundtrack BUT the game works , then you probably made a mistake when adding the assets to the APK

**Reply by TrexCM241** [reply]
so what can i do?

**Reply by TrexCM241** [reply]
maybe because I didn't use aapt.exe when copying music files?

**Comment by yosoul-0** [comment]
i have an esm-9110 and the analog didn't work. I've been trying for 1 hour

**Reply by MrPowerGamerBR** [reply]
Sadly I don't have that specific joystick to test with, are you using GameKeyboard+?

**Reply by yosoul-0** [reply]
yes , it is usb otg

**Comment by xcfgn** [comment]
Is there any way to port it to ps3

**Reply by MrPowerGamerBR** [reply]
Nope, GameMaker: Studio doesn't support exporting games to PS3.

**Comment by theone390** [comment]
its been a while, but would it be possible to port pizza tower using this?

**Reply by MrPowerGamerBR** [reply]
Only if Pizza Tower was made with GameMaker: Studio too, if it was, then it should be possible.

**Reply by theone390** [reply]
it was made in gamemaker studio. also. the zipsigner app got taken down so a new method would be useful

**Reply by elmemeee** [reply]
I used Termux for signing the file. You need to have the java developer kit (use pkg install openjdk-17 I think). Here are the commands (copy UndertaleWrapper.apk to the home folder. To do this go to configuration, enable the permission to access files. Open Termux, type cd {the path of UndertaleWrapper. In my case, /storage/emulated/0/Download/} and then type mv UndertaleWrapper.apk $HOME). 
Make sure to replace everything in between the {} symbols with what corresponds. Everything in between this symbol " Is a single command, do NOT paste the " symbols too, Simce they arent part of the command.
"cd"

"keytool -genkey -v -keystore {name of keystore}.keystore -alias {short name for your keystore} -keyalg RSA -keysize 2048 -validity 10000"
It Will ask you for a password. Make sure to remember it.


"jarsigner -verbose -sigalg SHA1withRSA -keystore {name of keystore, not alias. make sure to include the file extensión .keystore} UndertaleWrapper.apk {the alias you put Before}"
It Will ask you for the password you set Before. It Will also ask for other details, you can fill them If you want to. Now, the apk SHOULD be signed. I'm doing all These commands from memory so I could have done a typo or a mistake, but it is so you can get a general idea of what to do.
Later, move the app to your download folder like this
"mv UndertaleWrapper.apk /storage/emulated/0/Download/". Hope it helps. You can also sign it with a computer, but I'm not sure How to do it that way.

**Comment by BrexxIGuess** [comment]
I followed this tutorial, 10/10 everything worked. The Z, X, C, and everything worked but, how do I use my bluetooth controller's joystick to droidtale? I've been trying to find it but nothing I've done has worked yet, sadly.

**Reply by veeeanti** [reply]
GameKeyboard+ can remap your joysticks to what you need for them to be, just need to go to the advanced options in the app (very last choice on the first settings screen), and then look for enable left / right stick settings. Then you can remap the joystick to your dpad!

**Comment by WafeeyGitHub123** [comment]
Why on my phone crashing😓😓😓

**Reply by veeeanti** [reply]
What phone do you have, and does it crash when you have GameKeyboard+ running alongside it?

**Comment by LimePaint** [comment]
every time i try to edit the apk, it knows it's been edited.
otherwise it seems like a good runner for most game maker 1 games.

**Reply by veeeanti** [reply]
You do sign it after modifying the files, correct?

**Comment by foognocchie** [comment]
Link do ZipSigner está morto. Tem alguma outra sugestão?

**Reply by elmemeee** [reply]
(español, ojalá se entienda) Puedes reemplazar ZipSigner por Termux. En Termux puedes ocupar keytool y jarsigner.

**Comment by UzumakiParadox** [comment]
Hi! I wanted to port a different game, and I actually managed to do everything and install the app, but, when I tried to open it, it just shows the splash screen and immediately crashes. Do you know what to do?

And also I wanted to know if there's a way to change the apk icon and name (so it shows the ones from the game I'm porting)

Thanks in advance!

**Reply by veeeanti** [reply]
App icons are located in "res/drawable" folders in the apk file, and the apps name can be changed in two places - inside AndroidManifest.xml, and in "res/values/strings.xml."

**Comment by Chaos153426** [comment]
When I enter the game, the gamepad does not appear, it only appears outside the game. I have tried Game Keyboard and Gamepad and none of them appear when enter the game

**Comment by elmemeee** [comment]
I just finished doing this but when I open it, it shows a Splash screen and closets. I know this Is not supported anymore but I still would appreciate a Little help. There are some things I did in a different way but from what I know it shouldnt affect the ending result. I am using a spanish patch, Instead of using zipsigner which Is no longer available I signed it with Termux using keytool and jarsigner, I copied the music without aapt and Instead copied them to the assets folder the same way you copied the game.droid file. I think the problem here Is not using aapt, so I think using it should work. Also, I enabled the storage settings. I have tried this with android 10 and android 9, with both not working.

**Reply by elmemeee** [reply]
Tl;dr, game not working, pls help.

**Reply by elmemeee** [reply]
Also, typo, in the beginning it should say closes not closets

**Reply by elmemeee** [reply]
I fixed it, turns out I had to use v1.08

**Reply by Abdul-T-Albaker** [reply]
V1.08 of what??

**Comment by Manomanomano4k** [comment]
First of all thank you very much for this it works! 

And second where can I find the save files of the game?

**Comment by Qwruz** [comment]
I have been trying for several hours to map D-Pad Up to the arrow key up and D-Pad Down to arrow key down etc. But there is no key for arrow up. I am connected to a ps5 controller and using gamekeyboard+ to map the keys (Like you said) Z X and C are working perfectly and were easy to set up it is just the damn arrow keys. And because I can't map the arrow keys I can't get past the "Name the fallen human." Section because I can't move.

**Reply by elmemeee** [reply]
i would recommend using modded gamepad and then octopus to map the keys

**Comment by breadbear69** [comment]
Im using a ps3 controller and the z,x,c buttons work, but i cannot use directions. Is there a way to bind keyboard arrows to pads?

**Comment by Papyrusty** [comment]
I didn't do the music, and followed the guide twice, but when I try installing it, it says the package appears to be invalid

**Comment by veeeanti** [comment]
Hi! Thank you for this port of Undertale, I remember using it to play the game at school when I was in the 7th grade or something like that lol. In any case, I need to ask about the controls code, or rather how to add support for other input methods. I'm able to install this on the Xperia Play, but not control it completely. The d-pad works fine, but the only other key that works is the select button, and I believe it's due to the fact that it's called "KEY_ENTER" in the source code for the kernel. I was hoping you would know a bit more about something like this because it could just be a gamemaker thing that I could just change by editing the data.win / game.droid file. Unfortunately, the port is much too heavy on the Play and causes GameKeyboard+ to crash very shortly after starting the game.

**Reply by veeeanti** [reply]
I managed to figure it out myself using undertalemodtool, again - thank you so much for making this port!

**Reply by leon-sm** [reply]
Hey, can you tell me how you did it? Trying to figure out how to play with joy cons.

**Comment by leon-sm** [comment]
Thanks for the tutorial. Sadly, neither of the gamepad mapping apps work with newer android versions. Do you have an alternative that works with android 14?

**Reply by elmemeee** [reply]
You can use a utmt script to modify the data.win file and add touch controls directly to the game

**Reply by leon-sm** [reply]
Do you have a tutorial or something like that? Never used utmt before. I don't want touch controls, I just want to use my joy cons.

**Comment by veeeanti** [comment]
I'll help you out with this ASAP, I modified the controls for Undertale to
work with the Xperia Play's inputs so the joy cons shouldn't be too
difficult. UTMT is also extremely easy to use by the way, touch controls
can be added by just running a built in script. However, physical controls
mods are significantly more difficult.

**Reply by Abdul-T-Albaker** [reply]
Hey. I dont know if you are still checking these comments but I need your help.

I'm having the same issue as 2 other commentors. Where I tried to install the data.win file the exact steps as the tutorial but it just crashes

What do I do?

**Comment by veeeanti** [comment]
Shit, I'm so sorry - I haven't done anything related to Undertale lately as
I've been highly focused on Half-Life stuff and GoldSrc / Source Engine
modding lately. When I get off work this afternoon I'll look into what
you're needing regarding controller inputs and get back to you on it. Is
there anything I can contact you on?

**Reply by MrPowerGamerBR** [reply]
Just a FYI @leon-sm, after the advent of Undertale Mod Tool (this tutorial is oooold, I made it in 2015, and back in the day Undertale Mod Tool didn’t exist) there are now modded Undertale versions tailored for phone gameplay

Maybe try searching a bit for “better” modded versions for phones, iirc there are some in the moddb website

**Comment by bugtheultrabugger** [comment]
gamepad modded isn't working on my phone (A15 5G)

(aliás ótima loritta kkkskskskk)

**Comment by Kayno725** [comment]
For some reason it still doesnt let me download it

**Comment by Ali070112** [comment]
Can I port undertale yellow with this?

**Reply by MrPowerGamerBR** [reply]
You can as long as the game (in this case, UT: Yellow) used the same GameMaker: Studio version as Undertale, however I think that they didn't use the same version, considering that UT: Yellow came waaaay later.

**Comment by Abdul-T-Albaker** [comment]
Do other data.win files of other games affect the process? Or is this tutorial just too old for my phone?

**Reply by MrPowerGamerBR** [reply]
> Do other data.win files of other games affect the process?

What do you mean by this?

You can use the `data.win` of another game, however the game MUST use the same GameMaker: Studio version that Undertale used, if not the game will crash on launch. You can, however, get a copy of the same version that the game used and compile a Android game with it and use it on another game.

**Reply by Abdul-T-Albaker** [reply]
That it used at the time of the tutorial? Or the one it's currently using?

**Reply by Abdul-T-Albaker** [reply]
Also how do you check what game maker version it's using?

**Reply by Abdul-T-Albaker** [reply]
The game in question is "nidhogg"

There is a video of someone running it using your method, so it is possible.

I just couldn't for the life of me figure out how. It just crashes on launch

**Comment by ghost** [comment]
hi

**Comment by umastrodev12** [comment]
Adorei :3

**Comment by Redwolf150Xperia** [comment]
This is going to be a long shot but!

The Device I played this on is an xperia play (Android 2.3.4 and I came up whit a ram issue, whit 512mb of ram it would crash after a few transition, quick solution hit the home screen would dodge the crash, 

Is there a way to make the port less ram intensive or restrict it soo it clear itseltf more frequently?

**Comment by Tricket123** [comment]
Hey bro , can i do this with deltarune too? Like , im a modder , then , i want to port my mod for android , can you pls tell me if its possible to port deltarune mods for android quickly...?

**Comment by dorpew** [comment]
:3

**Comment by dorpew** [comment]
primeiro de 2026

**Comment by loldo-g** [comment]
W

**Comment by DuksCodeBR** [comment]
O butterscotch é incrível também, power fazendo história de novo ksksks

**Comment by Arthur744** [comment]
hi, so there are some visual bugs on 1.0.8 and I've been trying to figure out how to fix it. maybe I can fix them with Undertale mod tool?
